src/main/java/Duke.java

```diff
@@ -1,10 +1,238 @@
+                break;
+            case "mark":
+                showToUser(MESSAGE_DIVIDER, ERROR_MESSAGE_MARK);
+                break;
+
+            case "unmark":
+                showToUser(MESSAGE_DIVIDER, ERROR_MESSAGE_UNMARK);
+                break;
+            case "todo":
+                showToUser(MESSAGE_DIVIDER, ERROR_MESSAGE_TODO);
+                break;
+            case "deadline":
+                showToUser(MESSAGE_DIVIDER, ERROR_MESSAGE_DEADLINE);
+                break;
+            case "event":
+                showToUser(MESSAGE_DIVIDER, ERROR_MESSAGE_EVENT);
+                break;
+            default:
+                showToUser(MESSAGE_DIVIDER, errorMessage);
+                break;
+        }
+    }
+
+    private static void exit() {
+        isRunning = false;
+        exitMessage();
+    }
+
+    private static void unknownCommand() {
+        showToUser(MESSAGE_DIVIDER, MESSAGE_UNKNOWN_COMMAND, MESSAGE_COMMAND_LISTS);
+    }
+
+    private static void list() {
+        showToUser(MESSAGE_DIVIDER_LIST);
+        for (int i = 0; i < Task.numberOfTasks; i++) {
+            System.out.print((i + 1) + ".");
+            tasksList[i].printTask();
+        }
+        showToUser(MESSAGE_DIVIDER);
+    }
+
+    private static void add(String line) {
+        String newTaskType = line.split(" ")[0].toLowerCase();
+        String[] taskDescriptor = line.replace(newTaskType, "").trim().split("/");
+        switch (newTaskType) {
+            case "todo":
+                addTodo(taskDescriptor);
+                break;
+            case "deadline":
+                addDeadline(taskDescriptor);
+                break;
+            case "event":
+                addEvent(taskDescriptor);
+                break;
+            default:
+                break;
+        }
+    }
+
+    private static void addTodo(String[] taskDescriptor) {
+        try {
+            if (taskDescriptor[0].trim().isEmpty()) {
+                throw new DukeException("No Task Description");
+            }
+            tasksList[Task.numberOfTasks] = new Todo(taskDescriptor[0]);
+            addTaskMessage();
+        } catch (ArrayIndexOutOfBoundsException e) {
+            commandErrorHandler("todo");
+        } catch (DukeException e) {
+            commandErrorHandler(e.getMessage());
+        }
+
+    }
+
+    private static void addDeadline(String[] taskDescriptor) {
+        try {
+            if (taskDescriptor[0].trim().isEmpty()) {
+                throw new DukeException("Error: No Task Description");
+            }
+            tasksList[Task.numberOfTasks] = new Deadline(taskDescriptor[0], taskDescriptor[1].trim());
+            addTaskMessage();
+        } catch (ArrayIndexOutOfBoundsException e) {
+            commandErrorHandler("deadline");
+        } catch (DukeException e) {
+            commandErrorHandler(e.getMessage());
+        }
+    }
+
+    private static void addEvent(String[] taskDescriptor) {
```

Comment:
Try to avoid duplicating code for addEvent, addDeadline, addTodo. Consider using abstract classes?